beater/config/sampling.go

```diff
@@ -71,15 +72,26 @@ type TailSamplingPolicy struct {
 }
 
 func (c *TailSamplingConfig) Unpack(in *config.C) error {
+	var err error
+	defer func() {
+		if err != nil {
+			logger := logp.NewLogger(logs.Config)
+			logger.Errorf("failed to setup tail sampling: %v", err)
+			logger.Info("continuing with tail sampling disabled")
+			*c = TailSamplingConfig(defaultTailSamplingConfig())
+		}
+	}()
 	type tailSamplingConfig TailSamplingConfig
 	cfg := tailSamplingConfig(defaultTailSamplingConfig())
-	if err := in.Unpack(&cfg); err != nil {
-		return errors.Wrap(err, "error unpacking tail sampling config")
+	if err = in.Unpack(&cfg); err != nil {
+		err = errors.Wrap(err, "error unpacking config")
+		return nil
 	}
 	cfg.Enabled = in.Enabled()
 	*c = TailSamplingConfig(cfg)
```

Comment:
Should we avoid setting an invalid config completely?

Reply:
you mean calling `Validate()` on the cfg, before setting `*c` to be equal to it? Since the code was already doing that and, should `Validate()` fail, `c` is set to the default config, I don't think it's a big deal, but I'm happy to clean it up if you think we should avoid this.

Reply:
I was concerned about the bad tbs config affecting the downstream code flow and causing unexpected cases. Since earlier we were terminating the server and thus such cases might not have been handled. If you think it is not a concern or protected by some other code flow then it is okay with me.

Reply:
In the event `Validate()` fails (returns a non-nil error), the defer func defined above should set `c` to be the default config. `Validate()` isn't available on `cfg` (as mentioned above by @axw, it drops the methods of the `TailSamplingConfig` type). Because the default config has enabled=false, it would actually be good to see if that effects downstream code, because then the code isn't respecting the fact that tbs is disabled